api/v1alpha1/envoypatchpolicy_types.go

```diff
@@ -118,7 +118,8 @@ type JSONPatchOperation struct {
 	// +optional
 	From *string `json:"from,omitempty"`
 	// Value is the new value of the path location.
-	Value apiextensionsv1.JSON `json:"value"`
+	// +optional
+	Value apiextensionsv1.JSON `json:"value,omitempty"`
```

Comment:
1. Optional field should be pointer
2. It would be helpful to explain when the `Value` should be nil and when not
3. Could we add a CEL validation for when this should be nil or not?

Reply:
> 1. Optional field should be pointer
got it, will do.

> 2. It would be helpful to explain when the `Value` should be nil and when not
`Value` only required by add and replace operation, will update the doc

> 3. Could we add a CEL validation for when this should be nil or not?
Yes, it should be do able, let me look at it.

Thanks for the review!

Reply:
> 3. Could we add a CEL validation for when this should be nil or not?

emm..., I got a hard time with this. I give try on this:

```
// +kubebuilder:validation:XValidation:rule=" !has(self.value) || (self.op != 'add' && self.op != 'replace') ",message="The Value field is required by add and replace operation"

```

It always shows that the value field is undefined.

In the end, I found this:
https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/2876-crd-validation-expression-language/README.md#type-system-integration

'object' with x-kubernetes-preserve-unknown-fields: object / "message type", unknown fields are NOT accessible in CEL expression

Reply:
+1 to adding CEL as well as in the PR

Reply:
@arkodg @zhaohuabing it seems like CEL doesn't work here due to the `value` is an unstruct message, is it ok to do the validation in the go code?

Reply:
can we check if field is not `nil` if `op` is nor `move` ?

Reply:
if not possible, lets add a validation check https://github.com/envoyproxy/gateway/blob/241e83839acba4349c746acfd797869f40e37aca/internal/xds/translator/jsonpatch.go#L53 here ?

Reply:
> can we check if field is not `nil` if `op` is not `move` ?

we can't, the CEL doesn't know the field of `value`.

As you suggested, the go code validation is probably the right place